module-1/desgindoc.md

```diff
@@ -0,0 +1,89 @@
+# Overview
+Implement For You recommendation system, that will provide personal goods recomendations based on user preferences inside Goodsend app.
+
+Goodsend is a **social wishlist app** that lets users save products with a simple tap, organize them into collections or collages, share with friends, and explore products shared by others.
+
+DAU ~200 users
+
+Images of product:
+User profile          |     Explore (chronological) page |     Good(item) modal
+:-------------------------:|:-------------------------:|:-------------------------:
+![alt text](image.png)  |  ![alt text](image-1.png) |  ![alt text](image-2.png)
+
+
+
+# Motivation
+Currently, users can explore new goods from friends feed, chronological feed or one for all users curated weekly feed.
+
+**Hypothesis:** If we implement For You feed with personalized recommendations we will improve user experience in exploring new items, increase engagement, retention rate and session time.
+
+**Success metrics:**
+Increased engagement (DAU, session time, retention rate)
+
+# Requirements & Constraints
+Should work only on users with sufficient data(>10 goods, tbd). Cold-start problem is out of scope.
+
+- Users should receive a personalized feed with at least 5 recommended items per day, based on their past activity, saved products, and interaction with other users' collections.
+- **Cost Efficiency:** not crazy cost's in terms of compute(<$25-50/month), models desirably should run on CPU
+- **Scalability:** The recommendation system should be designed to scale as the user base grows, ensuring that the system remains performant even as new users and more complex datasets are added.
+
+Current goods count: ~100,000
+Current users count: ~5000
```

Comment:
where are those coming from?

Reply:
i run a query to count the users and goods in db :)